components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -284,6 +284,7 @@ protected boolean handlePostAuthentication(Event event, String userName, UserSto
                 newClaims.put(AccountConstants.ACCOUNT_UNLOCK_TIME_CLAIM, "0");
                 newClaims.put(AccountConstants.ACCOUNT_LOCKED_CLAIM, Boolean.FALSE.toString());
                 newClaims.put(AccountConstants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM, "0");
+                IdentityUtil.threadLocalProperties.get().put(AccountConstants.ADMIN_INITIATED, false);
```

Comment:
Where does this thread local get cleared

Reply:
FIxed : https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/61